shared/src/main/scala/squants/thermal/Temperature.scala

```diff
@@ -141,14 +141,24 @@ object Temperature extends Dimension[Temperature] with BaseDimension {
   def apply[A](n: A, scale: TemperatureScale)(implicit num: Numeric[A]) = new Temperature(num.toDouble(n), scale)
 
   def apply(s: String): Try[Temperature] = {
-    val regex = "([-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?) *°? *(f|F|c|C|k|K|r|R)".r
-    s match {
-      case regex(value, unit) => unit match {
-        case "f" | "F" => Success(Fahrenheit(value.toDouble))
-        case "c" | "C" => Success(Celsius(value.toDouble))
-        case "k" | "K" => Success(Kelvin(value.toDouble))
-        case "r" | "R" => Success(Rankine(value.toDouble))
-      }
+    val regex = if (squants.Platform.name == "native") {
+      "([-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?) *(__DEGREE__)? *(f|F|c|C|k|K|r|R)".r
```

Comment:
I'm assuming the native platform doesn't support the unicode degree symbol, which is what prompted this change. I'm wondering if we can factor out just that into a conditional to avoid copy/pasting the regex.

Something like:
```scala
val degSymbol = if (squants.Platform.name == "native") "__DEGREE__" else "°"
val regex = s"([-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?) *${degSymbol}? *(f|F|c|C|k|K|r|R)".r
```